test/query_snapshot_stream_manager_test.dart

```diff
@@ -2,37 +2,125 @@ import 'dart:async';
 
 import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:fake_cloud_firestore/fake_cloud_firestore.dart';
+import 'package:fake_cloud_firestore/src/mock_query_snapshot.dart';
 import 'package:test/test.dart';
 
+class User {
+  final String name;
+
+  User(this.name);
+
+  User.fromMap(Map<String, dynamic> map) : name = map['name'];
+
+  Map<String, dynamic> toFirestore() => {
+        'name': name,
+      };
+}
+
 void main() {
   late FakeFirebaseFirestore instance;
 
   setUp(() async {
     instance = FakeFirebaseFirestore();
-    await instance.collection('users').add({
-      'name': 'Bob',
-    });
-    await instance.collection('users').add({
-      'name': 'Marie',
-    });
   });
 
-  test('Should allow concurrent cache modifications', () {
-    final subscriptions = <StreamSubscription<QuerySnapshot>>[];
-    instance
-        .collection('users')
-        .snapshots()
-        .listen((QuerySnapshot<Map<String, dynamic>> snapshot) {
-      // Add a new listener to the same query while an update being fired. This
-      // used to trigger a concurrency error.
-      subscriptions.add(instance
+  group('cache modifications', () {
+    setUp(() async {
+      await instance.collection('users').add({
+        'name': 'Bob',
+      });
+      await instance.collection('users').add({
+        'name': 'Marie',
+      });
+    });
+
+    test('Should allow concurrent cache modifications', () {
+      final subscriptions = <StreamSubscription<QuerySnapshot>>[];
+      instance
           .collection('users')
           .snapshots()
-          .listen((QuerySnapshot<Map<String, dynamic>> snapshot) {}));
+          .listen((QuerySnapshot<Map<String, dynamic>> snapshot) {
+        // Add a new listener to the same query while an update being fired. This
+        // used to trigger a concurrency error.
+        subscriptions.add(instance
+            .collection('users')
+            .snapshots()
+            .listen((QuerySnapshot<Map<String, dynamic>> snapshot) {}));
+      });
+      // Fire a snapshot update.
+      instance.collection('users').add({
+        'name': 'Steve',
+      });
+    });
+  });
+
+  group('listener updates', () {
+    late CollectionReference collectionRef;
+    setUp(() {
+      collectionRef = instance.collection('users');
     });
-    // Fire a snapshot update.
-    instance.collection('users').add({
-      'name': 'Steve',
+
+    void validate() {
```

Comment:
Would you mind renaming this into something more meaningful? Maybe `validateSnapshotsWithoutConverter()`.